enhancements/network/multi-network-vrfs.md

```diff
@@ -0,0 +1,924 @@
+```
+
+In the above flow, all packets have the dest MAC address changed to be that of the OVN GR, and then sent on the patch port
+towards the OVN GR. With multiple networks, host access to cluster IP service flows will now be modified to be on a per
+cluster IP basis. For example, if we assume two services exist on two user defined namespaces with cluster IPs 10.96.0.5
+and 10.96.0.6. The flows would look like:
+
+```
+[root@ovn-worker ~]# ovs-ofctl dump-flows breth0 table=0 | grep 10.96
+ cookie=0xdeff105, duration=22226.373s, table=0, n_packets=41, n_bytes=4598, idle_age=19399,priority=500,ip,in_port=LOCAL,nw_dst=10.96.0.5 actions=ct(commit,table=12,zone=64001,nat(src=169.254.169.2))
+ cookie=0xdeff105, duration=22226.373s, table=0, n_packets=41, n_bytes=4598, idle_age=19399,priority=500,ip,in_port=LOCAL,nw_dst=10.96.0.6 actions=ct(commit,table=13,zone=64001,nat(src=169.254.169.2))
+```
+
+The above flows are now per cluster IP and will send the packet to different dispatch tables (one per network):
+
+```
+[root@ovn-worker ~]# ovs-ofctl dump-flows breth0 table=12
+ cookie=0xdeff105, duration=22266.310s, table=2, n_packets=41, n_bytes=4598, actions=mod_dl_dst:02:42:ac:12:00:05,output:"patch-breth0-net1"
+ 
+ [root@ovn-worker ~]# ovs-ofctl dump-flows breth0 table=13
+ cookie=0xdeff105, duration=22266.310s, table=2, n_packets=41, n_bytes=4598, actions=mod_dl_dst:02:42:ac:12:00:06,output:"patch-breth0-net2"
+```
+
+Furthermore, host networked pod access to services will be restricted to the network it belongs to. For more information
+see the [Host Networked Pods](#host-networked-pods) section.
+
+Additionally, in the case where there is hairpin service traffic to the host
+(Host->Service->Endpoint is also the host), the endpoint reply traffic will need to be distinguishable on a per network
+basis. In order to achieve this, each OVN GR’s unique masquerade IP will be leveraged.
+
+For service access towards KAPI/DNS or potentially other services on the cluster default network, there are two potential
+technical solutions. Assume eth0 is the pod interface connected to the cluster default network, and eth1 is connected to the
+user-defined primary network:
+
+1. Add routes for KAPI/DNS specifically into the pod to go out eth0, while all other service access will go to eth1.
+This will then just work normally with the load balancers on the switches for the respective networks.
+
+2. Do not send any service traffic out of eth0, instead all service traffic goes to eth1. In this case all service
+traffic is flowing through the user-defined primary network, where only load balancers for that network are configured
+on that network's OVN worker switch. Therefore, packets to KAPI/DNS (services not on this network) are not DNAT'ed at
+the worker switch and are instead forwarded onwards to the ovn_cluster_router_&lt;user-defined network>. This router is
```

Comment:
> are instead forwarded onwards to the ovn_cluster_router_&lt;user-defined network>

Wait, how will this work on flat L2 user defined networks ? We do not plan on having an `ovn-cluster-router` there, do we ? We connect the LS implementing the network directly to each node's GR. 

Would it be the corresponding GR that would send the traffic to each network's management port ? @trozet

Reply:
Ah, I think you've explained it in [this comment](https://github.com/openshift/enhancements/pull/1623#discussion_r1626616629).

Yeah, seems I had got it right.

Reply:
correct. The GR will do this in the layer 2 case. I'll make that more clear in the text.